src/components/Post.tsx

```diff
@@ -1,28 +1,55 @@
-import { api } from "../utils/api";
-import { PostForm } from "./PostForm";
+import { Skeleton, Text } from "@chakra-ui/react";
+import { ListBlogPost } from "../types/blogPost";
+import { PostCard } from "./PostCard";
 
-export const Posts = () => {
-  const posts = api.blogPost.getBlogPosts.useQuery({ page: 0, perPage: 10 });
-  const postss = posts.data?.data;
-  const postsss = postss?.map((post) => ({
-    title: post.title,
-    score: post.score,
-    categories: post.categories,
-    author: post.author.id,
-    id: post.id,
-  }));
+interface PostCardProps {
+  postsQuery: ListBlogPost[] | undefined;
+  paginationPage: number;
+  postsTotal: number | undefined;
+  isLoading: boolean;
+  error: any;
+}
+
+export const Posts = ({
+  postsQuery,
+  postsTotal,
+  isLoading,
+  error,
+}: PostCardProps) => {
+  if (isLoading)
+    return (
+      <>
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+        <Skeleton>wedwed</Skeleton>
+        <br />
+      </>
+    );
+  if (postsTotal === 0) return <Text>No blog posts</Text>;
+  if (error) return <Text>Error </Text>;
   return (
     <div>
-      {postsss &&
-        postsss.map(({ author, categories, id, score, title }) => (
-          <PostForm
-            key={id}
-            title={title}
-            author={author}
-            categories={categories}
-            score={score}
-          />
-        ))}
+      {postsQuery?.map(({ author, categories, score, title, id }) => (
+        <PostCard
+          key={id}
+          title={title}
+          author={author.id}
+          categories={categories}
+          score={score}
+        />
+      ))}
     </div>
   );
```

Comment:
Overall, this component renders quite a lot of things. For now render logic is small, but it will be getting bigger and bigger. I think you should split it to multiple components. All the components could be in direction called `PostList`

```
const PostList = () => {
  const isEmpty = posts.length === 0;

  if (isLoading) return <PostListLoader />
  if (isEmpty) return <PostListEmpty />
  if (isError) return <PostListError />

  return <Box as="ul">{posts.map(post => <Box as="li"><PostCard/ ><</Box>)}</Box>

}
```